core/src/main/java/org/apache/brooklyn/entity/group/DynamicClusterImpl.java

```diff
@@ -775,6 +779,10 @@ public Entity addNode(Location loc, Map<?,?> extraFlags) {
         entity.sensors().set(CLUSTER_MEMBER, true);
         entity.sensors().set(CLUSTER, this);
 
+        int nextClusterMemberId = getConfig(NEXT_CLUSTER_MEMBER_ID);
+        entity.sensors().set(CLUSTER_MEMBER_ID, nextClusterMemberId);
```

Comment:
I've implemented this [elsewhere](https://github.com/apache/incubator-brooklyn/blob/master/software/database/src/main/java/org/apache/brooklyn/entity/database/mysql/MySqlClusterImpl.java#L83) using an `AttributeSensor<Supplier<Integer>>` which [increments an `AtomicBoolean`](https://github.com/apache/incubator-brooklyn/blob/master/software/database/src/main/java/org/apache/brooklyn/entity/database/mysql/MySqlClusterImpl.java#L273-L280).